src/Planners/src/ConvexHullHelper.cpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * @file ConvexHullHelper.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <libqhullcpp/PointCoordinates.h>
+#include <libqhullcpp/Qhull.h>
+#include <libqhullcpp/QhullFacetList.h>
+
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/EigenHelpers.h>
+
+#include <BipedalLocomotion/Planners/ConvexHullHelper.h>
+
+using namespace BipedalLocomotion::Planners;
+
+// Implementation
+struct ConvexHullHelper::Impl
+{
+    iDynTree::VectorDynSize b;
+    iDynTree::MatrixDynSize A;
+};
+
+ConvexHullHelper::ConvexHullHelper()
+{
+    m_pimpl = std::make_unique<Impl>();
+}
+
+ConvexHullHelper::~ConvexHullHelper()
+{
+}
+
+bool ConvexHullHelper::buildConvexHull(const std::vector<iDynTree::VectorDynSize>& points)
+{
+    const std::size_t size = points.front().size();
+
+    // check if the size of the vectors are all the same
+    auto point = points.cbegin();
+    // it is useless to check the size of the first vector
+    std::advance(point, 1);
+    for (; point < points.end(); std::advance(point, 1))
+    {
+        if (size != point->size())
+        {
+            std::cerr << "[ConvexHullHelper::buildConvexHull] All the vectors should belong to the "
+                         "same vectorial space."
+                      << std::endl;
+            return false;
+        }
+    }
+
+    // the qhull object can be called only once
+    orgQhull::Qhull qhull;
+
+    // it seems that the pointCoordinates element cannot be cleaned so a new point coordinates has to be instantiate
+    orgQhull::PointCoordinates pointCooridinates;
```

Comment:
```suggestion
    orgQhull::PointCoordinates pointCoordinates;
```

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/51/commits/5153be5e9e2b25fc6c5ed40f25f53e2dc85dd572